code/obj/playable_piano.dm

```diff
@@ -207,6 +212,15 @@ TYPEINFO(/obj/player_piano)
 					return 1
 		return 0
 
+	proc/is_pulser_auto_linking(var/mob/M)
+		if(ispulsingtool(M.l_hand) && SEND_SIGNAL(M.l_hand, COMSIG_IS_PLAYER_PIANO_AUTO_LINKER_ACTIVE)) return TRUE
+		if(ispulsingtool(M.r_hand) && SEND_SIGNAL(M.r_hand, COMSIG_IS_PLAYER_PIANO_AUTO_LINKER_ACTIVE)) return TRUE
+		if(istype(M, /mob/living/silicon))
+			for(var/atom/A in M:module_states)
```

Comment:
cast the mob as a silicon, `:` usage is forbidden

Reply:
Removed semicolon operator. `module_states` is not a part of `/mob/living/silicon`, so checks are made separately for `/mob/living/silicon/robot` and `/mob/living/silicon/hivebot`.